tool-testsuite/test/org/antlr/v4/test/tool/ATNDescriber.java

```diff
@@ -0,0 +1,203 @@
+package org.antlr.v4.test.tool;
+
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.atn.*;
+
+import java.io.InvalidClassException;
+import java.util.List;
+
+/** Make human readable set of ints from serialized ATN like this (for debugging / testing):
+ *
+ * max type 1
+ * 0:TOKEN_START -1
+ * 1:RULE_START 0
+ * 2:RULE_STOP 0
+ * 3:BASIC 0
+ * 4:BASIC 0
+ * rule 0:1 1
+ * mode 0:0
+ * 0:'a'..128169
+ * 0->1 EPSILON 0,0,0
+ * 1->3 EPSILON 0,0,0
+ * 3->4 SET 0,0,0
+ * 4->2 EPSILON 0,0,0
+ * 0:0
+ */
+public class ATNDescriber {
```

Comment:
Great! It looks similar to improvements in one of my PR: https://github.com/antlr/antlr4/pull/3556

Reply:
woohoo!